client/client.go

```diff
@@ -29,30 +32,76 @@ type Client struct {
 	Services *Services
 	// this is set by table client multiplexer
 	ProjectId string
+	// List of enabled services per project
+	EnabledServices map[string]map[GcpService]struct{}
 }
 
 func NewGcpClient(log hclog.Logger, bo BackoffSettings, projects []string, services *Services) *Client {
-	return &Client{
-		projects: projects,
-		logger:   log,
-		backoff:  bo,
-		Services: services,
+	c := &Client{
+		projects:        projects,
+		logger:          log,
+		backoff:         bo,
+		Services:        services,
+		EnabledServices: make(map[string]map[GcpService]struct{}),
 	}
+	if len(projects) == 1 {
+		c.ProjectId = projects[0]
+	}
+	return c
 }
 
 func (c Client) Logger() hclog.Logger {
 	return c.logger
 }
 
+//revive:disable:modifies-value-receiver
+
 // withProject allows multiplexer to create a new client with given subscriptionId
 func (c Client) withProject(project string) *Client {
-	return &Client{
-		backoff:   c.backoff,
-		projects:  c.projects,
-		Services:  c.Services,
-		logger:    c.logger.With("project_id", project),
-		ProjectId: project,
+	c.logger = c.logger.With("project_id", project)
+	c.ProjectId = project
+	return &c
+}
+
+//revive:enable:modifies-value-receiver
+
+func (c *Client) configureEnabledServices() error {
+	var esLock sync.Mutex
+	g, ctx := errgroup.WithContext(context.Background())
+	for _, p := range c.projects {
+		project := p
+		g.Go(func() error {
```

Comment:
semaphore might be needed here + we should add similar logic to the ulimit stuff in the sdk as number of goroutines here can be high as projects can be in the tens of thousands.

Reply:
Adjusted in the spirit of the sdk.